vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/machinehealthcheck/machinehealthcheck.go

```diff
@@ -0,0 +1,225 @@
+		err = e2e.CreateMachineHealthCheck(workerMachine.Labels)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("with node-unhealth-conditions configmap", func() {
+		BeforeEach(func() {
+			unhealthyConditions := &conditions.UnhealthyConditions{
+				Items: []conditions.UnhealthyCondition{
+					{
+						Name:    "Ready",
+						Status:  "Unknown",
+						Timeout: "60s",
+					},
+				},
+			}
+			glog.V(2).Infof("Create node-unhealth-conditions configmap")
+			err := e2e.CreateUnhealthyConditionsConfigMap(unhealthyConditions)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should delete unhealthy machine", func() {
+			stopKubeletAndValidateMachineDeletion(workerNode, workerMachine, 2*time.Minute)
+		})
+
+		AfterEach(func() {
+			glog.V(2).Infof("Delete node-unhealth-conditions configmap")
+			err := e2e.DeleteUnhealthyConditionsConfigMap()
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	It("should delete unhealthy machine", func() {
+		stopKubeletAndValidateMachineDeletion(workerNode, workerMachine, 6*time.Minute)
+	})
+
+	AfterEach(func() {
+		waitForReadyWorkers(numberOfReadyWorkers)
+		deleteMachineHealthChecks()
+		deleteKubeletKillerPods()
+	})
+})
+
+func waitNodeUnhealthyCondition(workerNodeName string) {
+	client, err := e2e.LoadClient()
+	Expect(err).ToNot(HaveOccurred())
+
+	key := types.NamespacedName{
+		Name:      workerNodeName,
+		Namespace: e2e.TestContext.MachineApiNamespace,
+	}
+	node := &corev1.Node{}
+	glog.Infof("Wait until node %s will have 'Ready' condition with the status %s", node.Name, corev1.ConditionUnknown)
+	Eventually(func() bool {
+		err := client.Get(context.TODO(), key, node)
+		if err != nil {
+			return false
+		}
+		readyCond := conditions.GetNodeCondition(node, corev1.NodeReady)
+		glog.V(2).Infof("Node %s has 'Ready' condition with the status %s", node.Name, readyCond.Status)
+		return readyCond.Status == corev1.ConditionUnknown
+	}, e2e.WaitLong, 10*time.Second).Should(BeTrue())
+}
+
+func waitForReadyWorkers(numberOfReadyWorkers int) {
+	client, err := e2e.LoadClient()
+	Expect(err).ToNot(HaveOccurred())
+
+	listOptions := runtimeclient.ListOptions{
+		Namespace: e2e.TestContext.MachineApiNamespace,
+	}
+	listOptions.SetLabelSelector(fmt.Sprintf("%s=", e2e.NodeWorkerLabel))
+	workers := &corev1.NodeList{}
+	glog.V(2).Infof("Wait until the environment will have %d ready workers", numberOfReadyWorkers)
+	Eventually(func() bool {
+		err := client.List(context.TODO(), &listOptions, workers)
+		if err != nil {
+			return false
+		}
+
+		readyWorkers := 0
+		for _, w := range workers.Items {
+			readyCond := conditions.GetNodeCondition(&w, corev1.NodeReady)
+			if readyCond.Status == corev1.ConditionTrue {
+				readyWorkers++
+			}
+		}
+
+		glog.V(2).Infof("Number of ready workers %d", readyWorkers)
+		return readyWorkers == numberOfReadyWorkers
+	}, 15*time.Minute, 10*time.Second).Should(BeTrue())
+}
+
+func deleteMachineHealthChecks() {
```

Comment:
Deleting all HC resources might cause trouble when we start running multiple tests in parallel. Can you delete only the single HC that was created? `CreateMachineHealthCheck` is creating only one HC. So `deleteMachineHealthChecks` as its counterpart should then delete only the same HC resource that was created.

Reply:
done